core/src/main/java/org/elasticsearch/snapshots/SnapshotInfo.java

```diff
@@ -310,10 +473,8 @@ public void writeTo(StreamOutput out) throws IOException {
      * @param in stream input
      * @return deserialized snapshot info
      */
-    public static SnapshotInfo readSnapshotInfo(StreamInput in) throws IOException {
-        SnapshotInfo snapshotInfo = new SnapshotInfo();
-        snapshotInfo.readFrom(in);
-        return snapshotInfo;
+    public static SnapshotInfo readSnapshotInfo(final StreamInput in) throws IOException {
```

Comment:
Can you drop this method and just call the constructor instead? I don't think it buys anything.